src/price/feed.ts

```diff
@@ -117,70 +145,26 @@ export class PriceFeed {
     return new Decimal(BigInt(response.price.value), SUBGRAPH_PRICE_PRECISION);
   }
 
-  private async fetchEthPrice(): Promise<Decimal> {
-    try {
-      if (RaftConfig.isTestNetwork) {
-        return this.fetchStEthTestnetPrice();
-      }
-
-      return (await this.fetchSubgraphPrice('ETH')) ?? this.fetchStEthPriceFromBlockchain();
-    } catch {
-      return this.fetchStEthPriceFromBlockchain();
+  private getTokenCollateralConfig(token: Token) {
```

Comment:
Maybe
```suggestion
  private getTokenCollateralConfig(token: CollateralToken) {
```
and then check if the token is collateral/not R.